src/apps/external_executor/external_executor.cpp

```diff
@@ -129,6 +124,170 @@ namespace externalexecutor
       return nullptr;
     }
 
+    std::unordered_map<std::string, MapStrategyPtr> map_index_strategies;
+
+    void install_index_service(ccfapp::AbstractNodeContext& node_context)
+    {
+      auto executor_auth_policy = std::make_shared<ExecutorAuthPolicy>();
+      ccf::AuthnPolicies executor_only{executor_auth_policy};
+
+      // registers the index and will store the streamptr
+      auto install_and_subscribe_index =
+        [this, &node_context](
+          ccf::endpoints::CommandEndpointContext& ctx,
+          externalexecutor::protobuf::IndexInstall&& payload,
+          ccf::grpc::StreamPtr<externalexecutor::protobuf::IndexWork>&&
+            out_stream) -> ccf::grpc::GrpcAdapterStreamingResponse {
+        std::string strategy = payload.strategy_name();
+
+        auto it = map_index_strategies.find(strategy);
+        if (it != map_index_strategies.end())
+        {
+          return ccf::grpc::make_error(
+            GRPC_STATUS_ALREADY_EXISTS,
+            fmt::format("Strategy {} already exists", strategy));
+        }
+        auto executor_id = get_caller_executor_id(ctx);
+
+        auto ds = payload.data_structure();
+        // Mark default ds as MAP
+        IndexDataStructure data_structure = MAP;
+
+        if (ds == externalexecutor::protobuf::IndexInstall::PREFIX_TREE)
+        {
+          data_structure = PREFIX_TREE;
+        }
```

Comment:
If we reused the protobuf-generated enum type here, rather than redefining, then we could get a simpler default behaviour (ie - this code doesn't claim to set a default, _always_ gets it from the message. Then the message likely has some well-defined default).